public/app/features/alerting/unified/components/rule-editor/VizWrapper.tsx

```diff
@@ -1,136 +1,86 @@
   thresholds?: ThresholdsConfig;
   thresholdsType?: GraphTresholdsStyleMode;
   onThresholdsChange?: (thresholds: ThresholdsConfig) => void;
 }
 
-type PanelFieldConfig = FieldConfigSource<GraphFieldConfig>;
-
-export const VizWrapper = ({
-  data,
-  currentPanel,
-  changePanel,
-  thresholds,
-  thresholdsType = GraphTresholdsStyleMode.Line,
-}: Props) => {
-  const [options, setOptions] = useState<PanelOptions>({
-    frameIndex: 0,
-    showHeader: true,
-  });
-  const vizHeight = useVizHeight(data, currentPanel, options.frameIndex);
-  const styles = useStyles2(getStyles(vizHeight));
-
-  const [fieldConfig, setFieldConfig] = useState<PanelFieldConfig>(defaultFieldConfig(data, thresholds));
-
-  useEffect(() => {
-    setFieldConfig((fieldConfig) => ({
-      ...fieldConfig,
-      defaults: {
-        ...fieldConfig.defaults,
-        thresholds: thresholds,
-        unit: defaultUnit(data),
-        custom: {
-          ...fieldConfig.defaults.custom,
-          thresholdsStyle: {
-            mode: thresholdsType,
-          },
-        },
-      },
-    }));
-  }, [thresholds, setFieldConfig, data, thresholdsType]);
-
-  const context: PanelContext = useMemo(
-    () => ({
-      eventBus: appEvents,
-      canEditThresholds: false,
-      showThresholds: true,
-    }),
-    []
-  );
-
-  if (!options || !data) {
-    return null;
-  }
+/** The VizWrapper is just a simple component that renders either a table or a graph based on the type of data we receive from "PanelData" */
+export const VizWrapper = ({ data, thresholds, thresholdsType }: Props) => {
+  const styles = useStyles2(getStyles);
+  const isTimeSeriesData = isTimeSeriesFrames(data.series);
+  const thresholdsStyle = thresholdsType ? { mode: thresholdsType } : undefined;
+  const timeRange = {
+    from: data.timeRange.from.valueOf(),
+    to: data.timeRange.to.valueOf(),
+  };
 
   return (
     <div className={styles.wrapper}>
-      <div className={styles.buttonGroup}>
-        <PanelPluginsButtonGroup onChange={changePanel} value={currentPanel} />
-      </div>
-      <AutoSizer>
-        {({ width }) => {
-          if (width === 0) {
-            return null;
-          }
-          return (
-            <div style={{ height: `${vizHeight}px`, width: `${width}px` }}>
-              <PanelContextProvider value={context}>
-                <PanelRenderer
-                  height={vizHeight}
-                  width={width}
-                  data={data}
-                  pluginId={currentPanel}
-                  title="title"
-                  onOptionsChange={setOptions}
-                  options={options}
-                  fieldConfig={fieldConfig}
-                />
-              </PanelContextProvider>
-            </div>
-          );
-        }}
+      <AutoSizer disableHeight>
+        {({ width }) => (
+          <div style={{ width }}>
+            {isTimeSeriesData ? (
+              <GraphContainer
```

Comment:
It looks like `PanelRenderer` has built-in error handling. As we no longer use it we need to handle error on our own